### ISSUE TITLE
crash with `inproc` mode

### ISSUE THREAD

**Issue by navono** [zeromq/cppzmq#498]
OS: Win10
Version: cppzmq: 4.7.1

Follow the [pubsub_multithread_inproc.cpp](https://github.com/zeromq/cppzmq/blob/master/examples/pubsub_multithread_inproc.cpp) code, everything is OK, but if I move the subscriber to dll and when some conditions are met, exit the thread, then process crash.

I have a reproduce repo in [here](https://github.com/navono/zeroMQ-crach-with-inproc-in-dll). Any suggestions?

![mq1](https://user-images.githubusercontent.com/3746639/124277073-3911ef80-db77-11eb-8035-d32a856be6ca.PNG)

![mq2](https://user-images.githubusercontent.com/3746639/124277103-4202c100-db77-11eb-8619-d2c2bb31b584.PNG)

**Comment by navono** [comment]
Dig some info in stackoverflow, not very precisely, but I guess is that the `ctx` value passed cross module caused.
Currently, just change `inproc` to `tcp` also work(need change ctx io threads to non-zero value).

**Comment by jlucke86** [comment]
I was sure I read somewhere that inproc:// is not supported on windows. Is it now supported?

**Comment by aviallon** [comment]
@jlucke86 No, it's `ipc`, as mentioned here: http://api.zeromq.org/2-1:zmq-ipc  

> The inter-process transport is currently only implemented on operating systems that provide UNIX domain sockets.
